### ISSUE TITLE
Recheck containers after sleep on 'Unable to ping docker daemon' error

### ISSUE THREAD

**Issue by dadittoz** [jwilder/docker-gen#136]
I've had a weird issue with jwilder/nginx-proxy that didn't work properly up after a system restart. The proxy container had restart=always, no dependencies and was probably started first. Somehow docker 1.9.1 API returns '404 page not found' error at the very beginning. Probably the container was up before API become available.
```
dockergen.1 | 2015/11/23 19:38:48 error listing containers: API error (404): 404 page not found
dockergen.1 | 2015/11/23 19:38:48 Unable to ping docker daemon: API error (404): 404 page not found
dockergen.1 | 2015/11/23 19:38:58 Watching docker events
```

Now, when docker-gen receives an error on a ping request it just sleeps for 10 seconds. After that no check is being made for containers that happen to come up while it was sleeping. So it misses all the containers that have restart=always on boot. 

And this kinda sucks :) Have to add a dependency to nginx-proxy so it won't be started first or restart the proxy container manually after boot.

**Comment by thaJeztah** [comment]
Did this change in docker 1.9.1? If so, it may be worth reporting in the docker issue tracker, at least so the maintainers are able to have a look at it.

**Comment by dadittoz** [comment]
@thaJeztah I believe it is a some kind of race condition. I am not able to reproduce it 100%. Have to reboot the server for it to happen. I didn't have this issue on last reboot. Anyhow proper handling of situation when API becomes unavailable by adding containers rescan is what should docker-gen do.

**Comment by thaJeztah** [comment]
@dadittoz thanks! It sounds like a race indeed, just if that changed in 1.9.1, feel free to open an issue if it becomes problematic :+1:

**Comment by jwilder** [comment]
Fixed via #156
